### DISCUSSION TITLE
systemd service fails if deployment in additionalManifests fails

### DISCUSSION THREAD

**Discussion by agracey** [k3s-io/k3s#11381] [Q&A]
**Environmental Info:**
K3s Version: 
v1.30.3+ks31

Node(s) CPU architecture, OS, and Version: 
x86_64, SL Micro 6.0

Cluster Configuration:

Single node (but shouldn't matter)

**Describe the bug:**

If I create a broken deployment in `/var/lib/rancher/k3s/server/manifests` then install k3s, the `k3s-server` systemd service will go into a failed state and nothing that depends on it will start. 

**Expected behavior:**
The k3s-server service should be considered running even if a custom addOn fails.

**Actual behavior:**
The k3s-server service fails. 

**Additional context / logs:**
I'm using Edge Image Builder to build an appliance image.

**Comment by brandond** [comment]
> If I create a broken deployment in /var/lib/rancher/k3s/server/manifests then install k3s, the k3s-server systemd service will go into a failed state and nothing that depends on it will start.

That's not how that works.

Did you perhaps deploy something that is so broken it causes other core components to fail? Something that is replacing core RBAC or configuration in a way that prevents embedded controllers from functioning? Just placing an invalid manifest in that dir will not cause k3s to exit. You can definitely create, modify, or or delete things such that K3s will be broken, but this isn't unique to use of the manifests dir.

Please post the full logs showing what you're seeing, along with the manifest in question.
